server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java

```diff
@@ -272,14 +272,34 @@ protected Builder newBuilder(String name, Map<String, Object> params) {
         }
     }
 
-    public abstract static class AbstractGeometryFieldType<Parsed, Processed> extends MappedFieldType {
+    public abstract static class AbstractSearchableGeometryFieldType extends MappedFieldType {
```

Comment:
Since `GeoPointFieldMapper` doesn't inherit from `AbstractGeometryFieldMapper` I would prefer `AbstractSearchableGeometryFieldType` to be in a separate file. It's not longer confined to `AbstractGeometryFieldMapper` and its derivatives, so doesn't make much sense making it an inner class. It would be also good to add some javadocs to it explaining the purpose of this class for future readers. Basically, something as simple as "a base class for geometry types that supports shape query builder" should help.

Reply:
thanks @imotov that's done